java/src/main/java/ai/onnxruntime/OrtLoraAdapter.java

```diff
@@ -0,0 +1,162 @@
+
+  private boolean closed = false;
+
+  private OrtLoraAdapter(long nativeHandle) {
+    this.nativeHandle = nativeHandle;
+  }
+
+  /**
+   * Creates an instance of OrtLoraAdapter from a byte array.
+   *
+   * @param loraArray The LoRA stored in a byte array.
+   * @throws OrtException If the native call failed.
+   * @return An OrtLoraAdapter instance.
+   */
+  public static OrtLoraAdapter create(byte[] loraArray) throws OrtException {
+    return create(loraArray, null);
+  }
+
+  /**
+   * Creates an instance of OrtLoraAdapter from a byte array.
+   *
+   * @param loraArray The LoRA stored in a byte array.
+   * @param allocator optional allocator or null. If supplied, adapter parameters are copied to the
+   *     allocator memory.
+   * @throws OrtException If the native call failed.
+   * @return An OrtLoraAdapter instance.
+   */
+  static OrtLoraAdapter create(byte[] loraArray, OrtAllocator allocator) throws OrtException {
+    Objects.requireNonNull(loraArray, "LoRA array must not be null");
+    long allocatorHandle = allocator == null ? 0 : allocator.handle;
+    return new OrtLoraAdapter(
+        createLoraAdapterFromArray(OnnxRuntime.ortApiHandle, loraArray, allocatorHandle));
+  }
+
+  /**
+   * Creates an instance of OrtLoraAdapter from a direct ByteBuffer.
+   *
+   * @param loraBuffer The buffer to load.
+   * @throws OrtException If the native call failed.
+   * @return An OrtLoraAdapter instance.
+   */
+  public static OrtLoraAdapter create(ByteBuffer loraBuffer) throws OrtException {
+    return create(loraBuffer, null);
+  }
+
+  /**
+   * Creates an instance of OrtLoraAdapter from a direct ByteBuffer.
+   *
+   * @param loraBuffer The buffer to load.
+   * @param allocator optional allocator or null. If supplied, adapter parameters are copied to the
+   *     allocator memory.
+   * @throws OrtException If the native call failed.
+   * @return An OrtLoraAdapter instance.
+   */
+  static OrtLoraAdapter create(ByteBuffer loraBuffer, OrtAllocator allocator) throws OrtException {
+    Objects.requireNonNull(loraBuffer, "LoRA buffer must not be null");
+    if (loraBuffer.remaining() == 0) {
+      throw new OrtException("Invalid LoRA buffer, no elements remaining.");
+    } else if (!loraBuffer.isDirect()) {
+      throw new OrtException("ByteBuffer is not direct.");
+    }
+    long allocatorHandle = allocator == null ? 0 : allocator.handle;
+    return new OrtLoraAdapter(
+        createLoraAdapterFromBuffer(
+            OnnxRuntime.ortApiHandle,
+            loraBuffer,
+            loraBuffer.position(),
+            loraBuffer.remaining(),
+            allocatorHandle));
+  }
+
+  /**
+   * Creates an instance of OrtLoraAdapter.
+   *
+   * @param absoluteAdapterPath path to the adapter file that is going to be memory mapped.
+   * @throws OrtException If the native call failed.
+   * @return An OrtLoraAdapter instance.
+   */
+  public static OrtLoraAdapter create(String absoluteAdapterPath) throws OrtException {
```

Comment:
>absoluteAdapterPath [](http://example.com/codeflow?start=45&length=19)

I had to relax this requirement and allow relativer paths in other APIs.

Reply:
Fixed.